openstf/model/train.py

```diff
@@ -160,30 +243,93 @@ def train_model_for_specific_pj(
         model_trainer.confidence_interval,
     )
     train_predict = predictor.make_forecast(
-        train_data.iloc[:, 1:].drop("Horizon", axis=1, errors="ignore")
+        split_input_data["train_data"]
+        .iloc[:, 1:]
+        .drop("Horizon", axis=1, errors="ignore")
     )
     validation_predict = predictor.make_forecast(
-        validation_data.iloc[:, 1:].drop("Horizon", axis=1, errors="ignore")
+        split_input_data["validation_data"]
+        .iloc[:, 1:]
+        .drop("Horizon", axis=1, errors="ignore")
     )
     test_predict = predictor.make_forecast(
-        test_data.iloc[:, 1:].drop("Horizon", axis=1, errors="ignore")
+        split_input_data["test_data"]
+        .iloc[:, 1:]
+        .drop("Horizon", axis=1, errors="ignore")
     )
 
-    context.perf_meter.checkpoint("predicting")
+    return {
+        "train_predict": train_predict,
+        "validation_predict": validation_predict,
+        "test_predict": test_predict,
+    }
+
+
+def create_evaluation_figures_for_specific_pj(
+    model_trainer, split_input_data, split_predicted_data
+):
+    """Create figures of feature importance and data series of input and predicted data.
+
+    Args:
+        model_trainer (MLModelType): model trainer
+        split_input_data (dict): dict of cleaned data with features, split in train,
+            validation and test
+        split_predicted_data (dict): dict of predicted data, split in train, validation
+            and test
+
+    Returns:
+        tuple: tuple containing:
+            plotly.graph_objects.Figure: A treemap of the features.
+            dict: dict of line plots of of split input and predicted
+                data series per prediction horizon.
+    """
 
     # Create figures
     figure_features = plot_feature_importance(model_trainer.feature_importance)
     figure_series = {
         f"Predictor{horizon}": plot_data_series(
-            [train_data, validation_data, test_data],
-            [train_predict, validation_predict, test_predict],
+            list(split_input_data.values()),
+            list(split_predicted_data.values()),
             horizon,
         )
-        for horizon in train_data.Horizon.unique()
+        for horizon in split_input_data["train_data"].Horizon.unique()
     }
+    return figure_features, figure_series
+
+
+def evaluate_new_model_for_specific_pj(
+    pj,
+    context,
+    model_trainer,
+    split_input_data,
+    split_predicted_data,
+    compare_to_old=True,
+):
+    """Evaluate quality of new model with old model, save model, figures and send Teams
+    message.
```

Comment:
Should this function really do that much? Would it not be more SRP and flexible to use seperate function for these tasks?

Reply:
Agree, I was also not happy about the number of actions in one function. I split it up